code/game/objects/items/latexballoon.dm

```diff
@@ -47,7 +47,7 @@
 	return
 
 /obj/item/latexballon/attackby(obj/item/W as obj, mob/user as mob)
-	if (W.sharpness)
+	if (W.sharpness_flags & SHARP_BLADE || W.sharpness_flags & SHARP_TIP || W.sharpness_flags & HOT_EDGE)
```

Comment:
`if (W.sharpness_flags & (SHARP_BLADE|SHARP_TIP|HOT_EDGE))` should work, and  is nicer to read.